File: core/x/javaapi/codegen/src/test/resources/golden/expected/com/example/grts/CreateOrderInput.java
package com.example.grts;

import graphql.schema.GraphQLInputObjectType;
import viaduct.java.api.context.ExecutionContext;
import viaduct.java.api.globalid.GlobalID;
import viaduct.java.api.internal.InputBase;
import viaduct.java.api.internal.InternalContext;
import viaduct.java.api.reflect.CompositeField;
import viaduct.java.api.reflect.Field;
import viaduct.java.api.reflect.Type;
import viaduct.java.api.reflect.TypeFields;
import java.time.Instant;
import java.time.LocalDate;
import java.time.OffsetTime;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

public class CreateOrderInput extends InputBase {

    public static final Type<CreateOrderInput> Reflection = Type.ofClass(CreateOrderInput.class);

    public static final class Fields implements TypeFields<CreateOrderInput> {
        private Fields() {}

        public static final Field<CreateOrderInput> __typename =
                Field.of("__typename", Reflection);
                public static final Field<CreateOrderInput> buyerId =
                                Field.of("buyerId", Reflection);

                public static final CompositeField<CreateOrderInput, Color> color =
                                CompositeField.of("color", Reflection, Color.Reflection);

                public static final Field<CreateOrderInput> amounts =
                                Field.of("amounts", Reflection);

                public static final Field<CreateOrderInput> note =
                                Field.of("note", Reflection);

    }

    // Package-private: input GRTs are constructed only through the validating Builder or
    // by sibling GRTs in this package (nested-input wrapping). Tenants cannot construct
    // one directly, so a @oneOf input cannot bypass the builder's fail-fast validation.
    CreateOrderInput(InternalContext context, Map<String, Object> data, GraphQLInputObjectType graphQLInputObjectType) {
        super(context, data, graphQLInputObjectType);
    }

    /**
     * Returns whether this input contains a value for {@code field} after GraphQL
     * defaults are applied. Explicit {@code null} counts as present. An omitted field
     * with a schema default is present; an omitted field without a default is absent.
     */
    public boolean isPresent(Field<CreateOrderInput> field) {
        return isFieldPresent(field);
    }

        public String getBuyerId() {
            return get("buyerId");
        }

        public Color getColor() {
            return getEnum("color", Color.class);
        }

        public List<Double> getAmounts() {
            return getScalarList("amounts");
        }

        public String getNote() {
            return get("note");
        }


    public Builder toBuilder() {
        return new Builder(__context(), getGraphQLInputObjectType(), getInputData());
    }

    public static Builder builder(ExecutionContext context) {
        return new Builder(InternalContext.from(context));
    }

    public static class Builder {
        private final InternalContext __context;
        private final GraphQLInputObjectType graphQLInputObjectType;
        private final Map<String, Object> data = new LinkedHashMap<>();

        private Builder(InternalContext __context) {
            this.__context = __context;
            this.graphQLInputObjectType =
                    (GraphQLInputObjectType) __context.getSchema().getSchema().getType("CreateOrderInput");
        }

        private Builder(InternalContext context, GraphQLInputObjectType type, Map<String, Object> data) {
            this.__context = context;
            this.graphQLInputObjectType = type;
            this.data.putAll(data);
        }

                public Builder buyerId(String buyerId) {
                    data.put("buyerId", buyerId);
        return this;
                }

                public Builder color(Color color) {
                    data.put("color", color);
        return this;
                }

                public Builder amounts(List<Double> amounts) {
                    data.put("amounts", amounts);
        return this;
                }

                public Builder note(String note) {
                    data.put("note", note);
        return this;
                }


        public CreateOrderInput build() {
            return new CreateOrderInput(__context, new LinkedHashMap<>(data), graphQLInputObjectType);
        }
    }
}

File: core/tenant/codegen/src/test/resources/golden/expected/grts/CreateOrderInput.kt
@file:Suppress("warnings")

package viaduct.api.grts

import graphql.schema.GraphQLInputObjectType
import viaduct.apiannotations.InternalApi
import viaduct.api.context.ExecutionContext
import viaduct.api.internal.InputTypeFactory
import viaduct.api.internal.InputValueBuilder
import viaduct.api.internal.InternalContext
import viaduct.api.internal.internal
import viaduct.api.internal.InputLikeBase
import viaduct.api.types.Input

@OptIn(InternalApi::class)
class CreateOrderInput internal constructor(
    override val context: InternalContext,
    override val inputData: Map<String, Any?>,
    override val graphQLInputObjectType: GraphQLInputObjectType,
): InputLikeBase(), viaduct.api.types.Input {
    init {
       TODO()
    }

    val buyerId: kotlin.String get() = TODO()

    val color: viaduct.api.grts.Color? get() = TODO()

    val amounts: kotlin.collections.List<kotlin.Double>? get() = TODO()

    val note: kotlin.String? get() = TODO()


    fun toBuilder() = Builder(context, graphQLInputObjectType, this.inputData.toMutableMap())

    object of {
        operator fun invoke(context: ExecutionContext, block: Builder.() -> Unit): CreateOrderInput =
            Builder(context).apply(block).build()
    }

    class Builder internal constructor(
        override val context: InternalContext,
        override val graphQLInputObjectType: GraphQLInputObjectType,
        override val inputData: MutableMap<String, Any?> = TODO()
    ) : InputLikeBase.Builder(), InputValueBuilder<CreateOrderInput> {

        constructor(context: ExecutionContext): this(
            context.internal,
            InputTypeFactory.inputObjectInputType("CreateOrderInput", context.internal.schema),
            mutableMapOf()
        )

        init {
            TODO()
        }

                    fun buyerId(value: kotlin.String): Builder = TODO()

                    fun color(value: viaduct.api.grts.Color?): Builder = TODO()

                    fun amounts(value: kotlin.collections.List<kotlin.Double>?): Builder = TODO()

                    fun note(value: kotlin.String?): Builder = TODO()


        final override fun build(): CreateOrderInput = TODO()
    }

    @OptIn(viaduct.apiannotations.InternalApi::class)
    object Reflection : viaduct.api.reflect.Type<viaduct.api.grts.CreateOrderInput> {
        override final val name = "CreateOrderInput"
        override final val kcls = viaduct.api.grts.CreateOrderInput::class
    }
    object Fields : viaduct.api.reflect.TypeFields<viaduct.api.grts.CreateOrderInput> {
            final val __typename: viaduct.api.reflect.Field<viaduct.api.grts.CreateOrderInput> =
                viaduct.api.internal.FieldImpl("__typename", viaduct.api.grts.CreateOrderInput.Reflection)

            final val buyerId: viaduct.api.reflect.Field<viaduct.api.grts.CreateOrderInput> =
                viaduct.api.internal.FieldImpl("buyerId", viaduct.api.grts.CreateOrderInput.Reflection)

            final val color: viaduct.api.reflect.CompositeField<viaduct.api.grts.CreateOrderInput, viaduct.api.grts.Color> =
                viaduct.api.internal.CompositeFieldImpl("color", viaduct.api.grts.CreateOrderInput.Reflection, viaduct.api.grts.Color.Reflection)

            final val amounts: viaduct.api.reflect.Field<viaduct.api.grts.CreateOrderInput> =
                viaduct.api.internal.FieldImpl("amounts", viaduct.api.grts.CreateOrderInput.Reflection)

            final val note: viaduct.api.reflect.Field<viaduct.api.grts.CreateOrderInput> =
                viaduct.api.internal.FieldImpl("note", viaduct.api.grts.CreateOrderInput.Reflection)

    }

}